### PR TITLE
bugs related to delettion and shufflong of  steps in clickable demos

### PR DESCRIPTION
**Pull request by HussainKhanSherwani**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [x] I have carefully reviewed my code and believe it is final from my side.
- [x] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [x] I have tested the new feature or updates in both light and dark themes.
- [x] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [x] I have tested my changes on mobile screen sizes.
- [x] I have tested my changes on tablet screen sizes.
- [x] I have tested my changes on laptop screen sizes.

### Functional Testing
- [x] I have thoroughly tested the functionality of my changes.
- [x] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [x] Explain the task/bug you are solving
- [x] Show the changes in the UI
- [x] Show the changes in the codebase
- [x] Show the changes you tested to make sure it works
- [x] Show the changes in various screen sizes
- [x] Show the changes in both light and dark themes
videoURL:https://www.loom.com/share/301e4ff296e249eaa7fbcd181d09ac2e?sid=e474b860-d019-485c-8ef4-b7ff760cbbab

### CODE REVIEW

File: academy-ui/src/components/clickableDemos/Edit/useEditClickableDemo.ts [hunk 1]
```diff
@@ -75,8 +75,10 @@ export function useEditClickableDemo(space: Space, demoId: string | null) {
     setClickableDemo((prevClickableDemo) => {
       const stepIndex = prevClickableDemo.steps.findIndex((s) => s.id === stepUuid);
       const updatedSteps = [...prevClickableDemo.steps];
-      updatedSteps[stepIndex - 1].order = stepIndex;
-      updatedSteps[stepIndex].order = stepIndex - 1;
```
Comment (MSamiTariq) [inline academy-ui/src/components/clickableDemos/Edit/useEditClickableDemo.ts:79]: can you explain why does this logic fail, although it should work fine?

Comment (HussainKhanSherwani) [reply academy-ui/src/components/clickableDemos/Edit/useEditClickableDemo.ts:79]: The problem is that we cannot directly update the object properties using assignment operator like done here
